internal/controller/cluster_summary_controller.go

```diff
@@ -0,0 +1,78 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	logs "github.com/projectsveltos/libsveltos/lib/logsettings"
+	"github.com/projectsveltos/ui-backend/internal/server"
+
+	configv1alpha1 "github.com/projectsveltos/addon-controller/api/v1alpha1"
+)
+
+type ClusterSummaryReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *ClusterSummaryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+	logger.V(logs.LogInfo).Info("Reconciling")
+
+	clusterSummary := &configv1alpha1.ClusterSummary{}
+	if err := r.Get(ctx, req.NamespacedName, clusterSummary); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
if ClusterSummary is not found, then we need to call removeClusterSummary. 

This controller does now own the ClusterSummary (the addon-controller does). So it might very likely be that by the time this reconciliation is called on a deleted ClusterSummary, the ClusterSummary is already gone (I.e, the addon-controller has processed that, removed the finalizer and let the api-server remove the ClusterSummary instance).